### ISSUE TITLE
Firefox Extension Update

### ISSUE THREAD

**Issue by stefnotch** [lzacharkow/tabby-cat-support#18]
I've noticed that the Firefox extension of Tabby Cats has been last updated in 2017. Would it be possible to update the Firefox extension and get those neat new items that are in the Chrome version?

> Version
>     0.995
> Size
>     272.03 KB
> Last updated
>     2 years ago (Sep 7, 2017)
> -- https://addons.mozilla.org/en-US/firefox/addon/tabby-cat-friend/

**Comment by betterphp** [comment]
This would be great :D

**Comment by anadon** [comment]
I'd be interested in an update.  As is, I don't see any place to pay to unlock dogs.

**Comment by betterphp** [comment]
The Chrome extension is actually compatible with Firefox too, it's just not published on the store.

Here's some steps to get it working
1. Install the up to date version in Chrome
2. Copy the extension files to a new folder, for me this was the folder `/home/jacek/snap/chromium/common/chromium/Default/Extensions/mefhakmgclhhfbdadeojlkbllmecialg/1.5.0_0`
3. Zip the contents of the folder, `manifest.json` should be in the root of the zip file.
4. Upload to Mozilla for signing at https://addons.mozilla.org/en-GB/developers/addon/submit/distribution Make sure to select "On your own" otherwise it go for review and public release.
5. Once signed download the .xpi file from the "my addons" section.
6. Open it with Firefox and click install.
7. Enjoy up to date tabby friends :)

The upload and signing step can be bypassed by installing the addon in debug mode but it's gets removed when you restart firefox :(

**Comment by niktss** [comment]
The extension was updated today.

**Comment by anadon** [comment]
I don't see any update listed on the add-on page for Firefox.  The
tabbycats.club links to the Firefox add-on page, suggesting that source is
authoritative.

**Comment by niktss** [comment]
Huh. In about:addons, I thought “Last updated” did not also mean “Installed at”, because I had just installed this extension and it showed “Last updated Jun 2”.
Sad it appears to be abandonware, at least the Firefox version…

**Comment by moogsG** [comment]
> The Chrome extension is actually compatible with Firefox too, it's just not published on the store.
> 
> Here's some steps to get it working
> 
>     1. Install the up to date version in Chrome
> 
>     2. Copy the extension files to a new folder, for me this was the folder `/home/jacek/snap/chromium/common/chromium/Default/Extensions/mefhakmgclhhfbdadeojlkbllmecialg/1.5.0_0`
> 
>     3. Zip the contents of the folder, `manifest.json` should be in the root of the zip file.
> 
>     4. Upload to Mozilla for signing at https://addons.mozilla.org/en-GB/developers/addon/submit/distribution Make sure to select "On your own" otherwise it go for review and public release.
> 
>     5. Once signed download the .xpi file from the "my addons" section.
> 
>     6. Open it with Firefox and click install.
> 
>     7. Enjoy up to date tabby friends :)
> 
> 
> The upload and signing step can be bypassed by installing the addon in debug mode but it's gets removed when you restart firefox :(

This worked great, thanks!

**Comment by numbat77** [comment]
I'd love to see the Firefox version updated too! I'd be happy to assist with that if you'd like. :) 
Love the chrome extension, but I hate chrome! Please let me have cute seals in Firefox! 
Thanks!

**Comment by Corgi** [comment]
> The Chrome extension is actually compatible with Firefox too, it's just not published on the store.
> 
> Here's some steps to get it working
> 
>     1. Install the up to date version in Chrome
> 
>     2. Copy the extension files to a new folder, for me this was the folder `/home/jacek/snap/chromium/common/chromium/Default/Extensions/mefhakmgclhhfbdadeojlkbllmecialg/1.5.0_0`
> 
>     3. Zip the contents of the folder, `manifest.json` should be in the root of the zip file.
> 
>     4. Upload to Mozilla for signing at https://addons.mozilla.org/en-GB/developers/addon/submit/distribution Make sure to select "On your own" otherwise it go for review and public release.
> 
>     5. Once signed download the .xpi file from the "my addons" section.
> 
>     6. Open it with Firefox and click install.
> 
>     7. Enjoy up to date tabby friends :)

These instructions still work as of 4/26/22! Thanks!

**Comment by softlybutch** [comment]
> The Chrome extension is actually compatible with Firefox too, it's just not published on the store.
> 
> Here's some steps to get it working
> 
>     1. Install the up to date version in Chrome
> 
>     2. Copy the extension files to a new folder, for me this was the folder `/home/jacek/snap/chromium/common/chromium/Default/Extensions/mefhakmgclhhfbdadeojlkbllmecialg/1.5.0_0`
> 
>     3. Zip the contents of the folder, `manifest.json` should be in the root of the zip file.
> 
>     4. Upload to Mozilla for signing at https://addons.mozilla.org/en-GB/developers/addon/submit/distribution Make sure to select "On your own" otherwise it go for review and public release.
> 
>     5. Once signed download the .xpi file from the "my addons" section.
> 
>     6. Open it with Firefox and click install.
> 
>     7. Enjoy up to date tabby friends :)
> 
> 
> The upload and signing step can be bypassed by installing the addon in debug mode but it's gets removed when you restart firefox :(


Is it possible to do this without installing chrome?

**Comment by betterphp** [comment]
@softlybutch  I don't think so no, unless you can find a way to download the extension from the chrome store as a file instead of installing it.

You don't have to sign in to chrome or anything, I just installed it to get this addon then removed it straight away.

**Comment by stefnotch** [comment]
Since the author is not responding, and the files are publicly available: 
Here is a download of the Chrome extension

[2.0.1_0.zip](https://github.com/lzacharkow/tabby-cat-support/files/10872085/2.0.1_0.zip)

**Comment by stefnotch** [comment]
I went through the process, and here's a Firefox signed tabby cats extension. Rename it to .xpi, drag-n-drop it into your browser and then it should be possible to install it.

[9821f45ab5864836b75b-2.0.1.xpi.txt](https://github.com/lzacharkow/tabby-cat-support/files/11786390/9821f45ab5864836b75b-2.0.1.xpi.txt)

**Comment by cyborgthegreat** [comment]
Hi! I tried these steps but mozilla isn't letting me install it because "manifest.json was not found". When I open the file, manifest.json is there (I used stefnotch's file). And I also can't find any my addon section, but that might be because I havent gotten to that step yet. Please help! Thanks!

**Comment by Namyts** [comment]
I have just followed betterphp's steps. Here is a zip which I uploaded (and am now using) for 3.0.0
(Only change was some tweaks to the `manifest.json`)

[tcu.zip](https://github.com/user-attachments/files/31123709/tcu.zip)
